extensions/amp-analytics/0.1/events.js

```diff
@@ -421,22 +415,22 @@ export class VisibilityTracker extends EventTracker {
 
     // An AMP-element. Wait for DOM to be fully parsed to avoid
     // false missed searches.
-    const unlistenPromise = this.root.ampdoc.whenReady().then(() => {
-      const selectionMethod = config['selectionMethod'] ||
+    const selectionMethod = config['selectionMethod'] ||
           visibilitySpec['selectionMethod'];
-      const element = user().assertElement(
-          this.root.getAmpElement(
-              (context.parentElement || context),
-              selector,
-              selectionMethod),
-          `Element "${selector}" not found`);
-      return visibilityManager.listenElement(
-          element,
-          visibilitySpec,
-          this.getReadyPromise(waitForSpec, selector, element),
-          createReadyReportPromiseFunc,
-          this.onEvent_.bind(this, eventType, listener, element));
-    });
+    const unlistenPromise = this.root.getAmpElement(
+        (context.parentElement || context),
+        selector,
+        selectionMethod
+        ).then(element => {
+          element =
```

Comment:
I checked we always check if element is not null. Hence I moved to existing APIs. PTAL